File: leetcode/c/007.c
/* 007.Reverse Integer */

#include<stdio.h>

/* A recursion to calculate reverse number. */
int _reverse(int x, int r) {
    if (x == 0)
        return r;
    if ((r * 10 / 10) != r)
        return 0;
    int re = (r * 10) + (x % 10);
    if ((r > re && r > 0) || (r < re && r < 0))
        return 0;
    return _reverse(x / 10, re);
}

int reverse(int x) {
    return _reverse(x, 0);
}

int main() {
    printf("%d\n", reverse(1234578));
    return 0;
}

File: leetcode/py/007.py
# 007. Reverse Integer
import unittest

class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 0:
            x = -x
            print(x)
            # my_string[::-1] is used to reverse string.
            x = -int(str(x)[::-1])
        else:
            x = int(str(x)[::-1])

        if (x > 2147483647) or (x < -2147483648):
            return 0
        else:
            return x

class SolutionUnitTest(unittest.TestCase):
    def setup(self):
        pass
    def tearDown(self):
        pass
    def testsingleNumber(self):
        # data = [0,1,3,2]
        s = Solution()
        self.assertEqual(s.reverse(123456789), 987654321)

if __name__ == '__main__':
    unittest.main()
